File: app/src/main/java/com/doubean/ford/data/vo/GroupDetail.java
package com.doubean.ford.data.vo;

import android.graphics.Color;
import android.text.TextUtils;

import androidx.annotation.NonNull;
import androidx.room.Ignore;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

public class GroupDetail implements Serializable {
    @NonNull
    public String id;

    @NonNull
    public String name;

    @SerializedName("member_count")
    public int memberCount;

    @SerializedName("topic_count")
    public int postCount;

    @SerializedName("create_time")
    public LocalDateTime dateCreated;

    @SerializedName("sharing_url")
    public String url;

    @SerializedName("avatar")
    public String avatarUrl;

    @SerializedName("member_name")
    public String memberName;

    @SerializedName("desc")
    public String description;

    @SerializedName("group_tabs")
    public List<GroupTab> tabs;

    @SerializedName("background_mask_color")
    public String colorString;

    public String uri;

    @Ignore
    public int getColor() {
        return Color.parseColor(TextUtils.isEmpty(colorString) ? "#FFFFFF" : colorString);
    }

    @Ignore
    public GroupItem toGroupItem() {
        GroupItem groupItem = new GroupItem();
        groupItem.name = this.name;
        groupItem.url = this.url;
        groupItem.dateCreated = this.dateCreated;
        groupItem.avatarUrl = this.avatarUrl;
        groupItem.shortDescription = this.description;
        groupItem.id = this.id;
        groupItem.postCount = this.postCount;
        return groupItem;
    }
}

File: app/src/main/java/com/doubean/ford/data/vo/GroupDetail.kt
package com.doubean.ford.data.vo

import android.graphics.Color
import android.text.TextUtils
import androidx.room.*
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime


data class GroupDetail(
    val id: String,
    val name: String,

    @SerializedName("member_count")
    val memberCount: Int = 0,

    @SerializedName("topic_count")
    val postCount: Int = 0,

    @SerializedName("create_time")
    val dateCreated: LocalDateTime? = null,

    @JvmField
    @SerializedName("sharing_url")
    val url: String,

    @SerializedName("avatar")
    val avatarUrl: String? = null,

    @SerializedName("member_name")
    val memberName: String? = null,

    @SerializedName("desc")
    val description: String? = null,

    @SerializedName("group_tabs")
    val tabs: List<GroupTab>? = null,

    @SerializedName("background_mask_color")
    val colorString: String? = null,

    val uri: String
) : java.io.Serializable {

    val color: Int
        get() = Color.parseColor(if (TextUtils.isEmpty(colorString)) "#FFFFFF" else colorString)

    @Ignore
    fun toGroupItem() = GroupItem(
        name = name,
        url = url,
        uri = uri,
        dateCreated = dateCreated,
        avatarUrl = avatarUrl,
        shortDescription = description,
        id = id,
        postCount = postCount
    )
}

    
